src/integ_test_resources/ios/sdk/integration/cdk/cdk_integration_tests_ios/iot_stack.py

```diff
@@ -205,29 +220,126 @@ def setup_custom_authorizer(self):
+            "Sample custom authorizer that allows or denies based on username and password",
+            {"custom_auth_user_pass_username": Constants.custom_auth_user_pass_username,
+             "custom_auth_user_pass_password": Constants.custom_auth_user_pass_password},
+            "us-east-1",
+        )
+        create_authorizer_policy = aws_iam.PolicyStatement(
+            effect=aws_iam.Effect.ALLOW,
+            actions=[
+                "iot:CreateAuthorizer",
+                "iot:UpdateAuthorizer",
+                "iot:DeleteAuthorizer",
+                "iot:UpdateDomainConfiguration",
+                "iot:CreateDomainConfiguration",
+                "iot:DescribeDomainConfiguration",
+                "iot:DeleteDomainConfiguration",
+            ],
+            resources=["*"],
+        )
+        provider_lambda = aws_lambda.SingletonFunction(
+            self,
+            "iot_custom_authorizer_user_pass_provider_lambda",
+            uuid="iot_custom_authorizer_user_pass_provider_lambda_20200727123737",
+            runtime=aws_lambda.Runtime.PYTHON_3_7,
+            code=aws_lambda.Code.asset("custom_resources/iot_custom_authorizer_user_pass_provider"),
+            handler="iot_custom_authorizer_user_pass_provider.on_event",
+            description="Sets up an IoT custom authorizer for user password & required domain config due to beta status",
+            environment={"custom_auth_user_pass_uuid" : Constants.custom_auth_user_pass_uuid,
+                         "custom_auth_user_pass_default_authorizer_name" : Constants.custom_auth_user_pass_default_authorizer_name,
+                         "custom_auth_user_pass_domain_configuration_name" : Constants.custom_auth_user_pass_domain_configuration_name},
+            current_version_options=aws_lambda.VersionOptions(
+                removal_policy=core.RemovalPolicy.DESTROY
+            ),
+            initial_policy=[create_authorizer_policy],
+        )
+
+        provider = custom_resources.Provider(
+            self, "iot_custom_authorizer_user_pass_provider", on_event_handler=provider_lambda
+        )
+
+        public_key = iot_custom_authorizer_key_resource.get_att(
+            "custom_authorizer_public_key"
+        ).to_string()
+
+        iot_endpoint = core.CustomResource(
+            self,
+            "iot_custom_authorizer_user_pass",
+            resource_type="Custom::IoTCustomAuthorizer",
+            service_token=provider.service_token,
+            properties={
+                "authorizer_function_arn": authorizer_function_arn,
+                "authorizer_name": custom_authorizer_name,
+                "public_key": public_key,
+                "token_key_name": token_key_name,
+            },
+        )
+        endpoint_address = iot_endpoint.get_att("BetaEndpointAddress").to_string()
+        self._parameters_to_save["iot_beta_endpoint_address"] = endpoint_address
+
+    def merge_dict(self, dict1, dict2):
+        res = {**dict1, **dict2}
```

Comment:
nit: if this isn't using any instance members of the enclosing class, consider making it a static method. It probably doesn't really belong in the iot_stack anyway, since it's more of a utility function. And see below--you may not even need it if you use `update` :)

Reply:
Moved to update, thanks!